spec/models/user_spec.rb

```diff
@@ -0,0 +1,22 @@
+require "rails_helper"
+
+describe User do
+  describe "validations" do
+    it { is_expected.to validate_presence_of(:email) }
+    it { is_expected.to validate_presence_of(:username) }
+
+    it "validates uniqueness of the user's email" do
+      create(:user, email: "geek@geek.com")
+      user_2 = build(:user, email: "geek@geek.com")
```

Comment:
even better, something like `duplicate_user` or `user_with_duplicate_email` makes it really obvious what this user is and why we expect it not to be valid